demos/examples/UnbiasedCompetitionPNAS2019_tutorial/getPoissonDutyCycleGating.m

```diff
@@ -0,0 +1,126 @@
+function [S,dynrate,spikeevents] = getPoissonDutyCycleGating(tOn,tOff,latency_,freq_,normFreqSigma_,phase_,widthSigma,baseline,dc_,ac_,slopeDutyCycle_,thresholdDutyCycle_,tau,kick,T,N)
+
+  if nargin<1,  tOn = 0;                  end
+  if nargin<2,  tOff = 0;                 end
+  if nargin<3,  latency_ = 0;             end
+  if nargin<4,  freq_ = 0;                end
+  if nargin<5,  normFreqSigma_ = 0.03;    end
+  if nargin<6,  phase_ = 0;               end
+  if nargin<7,  widthSigma = 0.001;       end % 0.001 represents an abrupt connectivity transition (in contrast to 0.1; it only applies to dc+ac not the baseline)
+  if nargin<8,  baseline = 0;             end % Hz
+  if nargin<9,  dc_ = 0;                  end % Hz
+  if nargin<10, ac_ = 0;                  end % Hz
+  if nargin<11, slopeDutyCycle_ = 1;      end % in ms
+  if nargin<12, thresholdDutyCycle_ = 0;  end
+  if nargin<13, tau = 2;                  end
+  if nargin<14, kick = 1;                 end
+  if nargin<15, T = (0:.01:1000)';        end
+  if nargin<16, N = 1;                    end
```

Comment:
[Function argument validation](https://es.mathworks.com/help/matlab/matlab_prog/function-argument-validation-1.html) can help managing optional and name-value arguments.

Reply:
This is a good suggestion for future iterations and models.